crucible/src/Lang/Crucible/Simulator/RegMap.hs

```diff
@@ -141,17 +144,26 @@ muxAny s itefns p (AnyValue tpx x) (AnyValue tpy y)
   | Just Refl <- testEquality tpx tpy =
        AnyValue tpx <$> muxRegForType s itefns tpx p x y
   | otherwise =
-    addFailedAssertion s $
```

Comment:
What is the rationale for using `throwIO` over `addFailedAssertion` here (and in other places in this commit)?

Reply:
I'm also wondering about this.  Having to separately call `getCurrentProgramLoc` + `throwIO` seems a little annoying.

Reply:
`addFailedAssertion` requires the "backend" not just the "sym", and I'm not entirely convinced that adding a proof obligation for situations where the simulator doesn't support some feature is semantically correct anyways.

Maybe it's better to make a little combinator that makes this pattern more concise.